enhancements/insights/config-api.md

```diff
@@ -0,0 +1,277 @@
+- Define a new configuration option providing a way to trigger on demand Insights gathering
+- Define a new configuration option providing a way to include/exclude particular Insights data gatherers
+- Provide Insights status & reporting in the status field in the new API
+
+### Non-Goals
+
+TBD
+
+## Proposal
+
+The proposal is to introduce a new Insights `CustomResourceDefinition` in the OpenShift API providing a way to configure the Insights operator and to report its status. The new CRD should follow a common CRD structure providing `spec` and `status` attributes. 
+
+```go
+type InsightsSpec struct {
+	// Gathering spec attribute includes all the configuration options related to
+	// gathering of the Insights archive and its uploading to the ingress.
+	Gathering Gathering `json:"gathering,omitempty"`
+	// SimpleContentAcces spec attribute includes all the configuration options related to
+	// periodical downloading of the SCA certificates.
+	SimpleContentAccess SimpleContentAccess `json:"simpleContentAccess,omitempty"`
+	// ClusterTransfer spec attribute includes all the configuration options related to
+	// periodical check of pending cluster transfers.
+	ClusterTransfer ClusterTransfer `json:"clusterTransfer,omitempty"`
+}
+
+type InsightsStatus struct {
+	// GatheringStatus provides basic information about the last Insights gathering.
+	GatheringStatus GatheringStatus `json:"gatheringStatus,omitempty"`
+	// SimpleContentAccessStatus provides basic information about the SCA controller.
+	SimpleContentAccessStatus SimpleContentAccessStatus `json:"scaStatus,omitempty"`
+	// ClusterTransferStatus provides basic information about the ClusterTransfer controller.
+	ClusterTransferStatus ClusterTransferStatus `json:"clusterTransferStatus,omitempty"`
+	// ReportStatus provides general Insights analysis results.
+	ReportStatus ReportStatus `json:"insightsReport,omitempty"`
+}
+
+type Gathering struct {
+	// Interval defines the time frequency of gathering. The duration is 2h by default.
+	Interval time.Duration `json:"interval,omitempty"`
+	// UploadEndpoint defines ingress endpoint for uploading Insights archive
+	UploadEndpoint string `json:"endpoint,omitempty"`
+	// EnableObfuscation is flag enabling the global obfuscation of the IP addresses and base domain
+	// in the Insights archive data.
+	EnableObfuscation bool `json:"enableObfuscation,omitempty"`
+	// Enables user to force Insights data gathering by setting a new reason
+	ForceGatheringReason string `json:"forceGatheringReason,omitempty"`
+	// List of gatherers to be included in the gathering. Empty list means that all gatherers are included
+	Gatherers []string `json:"gatherers,omitempty"`
+	// InsightsReport includes all the configuration options related to downloading the Insights report.
+	InsightsReport InsightsReport `json:"insightsReport,omitempty"`
+	// Flag to disable gathering.
+	Disabled bool `json:"disabled,omitempty"`
```

Comment:
No. This is intended to completely disable gathering. However, we should say that if it's disabled then it can't be forced.

Reply:
Why can't it be forced as a one-off? I don't normally want my cluster phoning home but I'm talking to support and they think we'd make more progress if my cluster sent in an Insights report, how do we facilitate that? Toggle disabled back and forth?

Reply:
@sdodson I think I don't understand your questions. This is an option how to disable Insights data gathering in a cluster and thus sending any data to RH. 
Of course, we want most clusters to stay connected (to be able to provide support/recommendations), but there are some exceptions. Currently the only way how to disable it is to remove the corresponding token from the `pull-secret`, which is very incovenient, because it also disables telemetry AFAIK.

Reply:
Yeah, it's fine to include the ability to disable the gathering. I was mostly following on Ben's question of how it interacts with `ForceGatherReason`, I can certainly envision scernarios where an admin doesn't want a cluster sending in Insights data but on a one-off does. I was pondering if allowing `ForceGatherReason` to supersede `Disabled` may be one way to achieve that.

I don't care how it's implemented, either by obeying `ForceGatherReason` when disabled or by momentarily (how long?) toggling `Disabled`.  Either way I think one of the fields needs to make it clear whether or not `ForceGatherReason` is ignored when `Disabled` is true because `Force` generally implies an override, to me at least.

Reply:
OK thanks. I see. I think I don't have strong preference here but I agree that it needs to be clear. Superseding the disabled state is definitely an option and probably better. It's true that `Force` implies an override.